transit_odp/organisation/tests/csv/test_service_codes.py

```diff
@@ -121,92 +138,308 @@ def test_registered_service_code():
+        '"Scope Status"',
+        '"Seasonal Status"',
+        '"Staleness Status"',
+        '"Licence Number"',
+        '"Service Code"',
+        '"Line"',
+        '"Dataset ID"',
+        '"Effective Seasonal Start Date"',
+        '"Seasonal Start Date"',
+        '"Seasonal End Date"',
+        '"Effective stale date due to end date"',
+        '"Effective stale date due to Effective last modified date"',
+        '"Last modified date < Effective stale date due to OTC effective date"',
+        '"Effective stale date due to OTC effective date"',
     ]
-    assert call_args_list[1] in expected_missing_rows
-    assert call_args_list[2] in expected_missing_rows
-    assert call_args_list[3] == call(
-        (
-            ServiceCodesCSV.REGISTERED,
-            licence_number,
-            service_codes[0],
-            service_numbers[0],
-            str(dataset1.id),
-        )
-    )
+
+    assert csv_output["row1"][0] == '"No"'
+    assert csv_output["row1"][1] == '"Unpublished"'
+    assert csv_output["row1"][2] == '"Registered"'
+    assert csv_output["row1"][3] == '"In Scope"'
+    assert csv_output["row1"][4] == '"Out of Season"'
+    assert csv_output["row1"][5] == '"Not Stale"'
+    assert csv_output["row1"][14] == '""'
+    assert csv_output["row1"][15] == '"True"'
+    assert csv_output["row1"][16] == '"2023-05-12"'
+
+    assert csv_output["row2"][0] == '"Yes"'
+    assert csv_output["row2"][1] == '"Unpublished"'
+    assert csv_output["row2"][2] == '"Registered"'
+    assert csv_output["row2"][3] == '"In Scope"'
+    assert csv_output["row2"][4] == '"In Season"'
+    assert csv_output["row2"][5] == '"Not Stale"'
+    assert csv_output["row2"][14] == '""'
+    assert csv_output["row2"][15] == '"True"'
+    assert csv_output["row2"][16] == '"2023-05-12"'
+
+    assert csv_output["row3"][0] == '"No"'
+    assert csv_output["row3"][1] == '"Unpublished"'
+    assert csv_output["row3"][2] == '"Registered"'
+    assert csv_output["row3"][3] == '"Out of Scope"'
+    assert csv_output["row3"][4] == '""'
+    assert csv_output["row3"][5] == '"Not Stale"'
+    assert csv_output["row3"][14] == '""'
+    assert csv_output["row3"][15] == '"True"'
+    assert csv_output["row3"][16] == '"2023-05-12"'
+
+    assert csv_output["row4"][0] == '"Yes"'
+    assert csv_output["row4"][1] == '"Published"'
+    assert csv_output["row4"][2] == '"Registered"'
+    assert csv_output["row4"][3] == '"In Scope"'
+    assert csv_output["row4"][4] == '"In Season"'
+    assert csv_output["row4"][5] == '"Stale - OTC Variation"'
+    assert csv_output["row4"][14] == '"2023-06-24"'
+    assert csv_output["row4"][15] == '"False"'
+    assert csv_output["row4"][16] == '"2023-02-10"'
+
+    assert csv_output["row5"][0] == '"Yes"'
+    assert csv_output["row5"][1] == '"Published"'
+    assert csv_output["row5"][2] == '"Registered"'
+    assert csv_output["row5"][3] == '"In Scope"'
+    assert csv_output["row5"][5] == '"Stale - End date passed"'
```

Comment:
I can see in a few cases you're not testing the value of Seasonal Status. I think that's because there's a bug in your code that writes it!

In this case, and the cases below, it looks like you've created a seasonal service and so I'd expect a status of Out of Season.